### PR TITLE
Add annual well care visit to performance dashboard

### PR DESCRIPTION
**Pull request by gigxz**

### CODE REVIEW

File: drivers/claims_reporting/app/models/claims_reporting/medical_claim.rb [hunk 1]
```diff
@@ -41,11 +44,14 @@ class MedicalClaim < HealthBase
...
+    scope :annual_well_care_visits, -> do
+      regex_str = "^(#{ANNUAL_WELLCARE_DIAGNOSIS_CODE_PREFIXES.join('|')})"
+      matching_icd10cm(regex_str).or(where(icd_version: 10, procedure_code: ANNUAL_WELLCARE_PROCEDURE_CODES))
+    end
+
     scope :matching_icd10cm, ->(pg_regexp_str) do
-      # Tip: pg_trgm gist or gin index can make ~ operators fast
-      # this will otherwise be slow!
-      where <<~SQL.squish, pattern: pg_regexp_str
-        COALESCE(icd_version. '10') AND (
```
Comment (gigxz) [inline drivers/claims_reporting/app/models/claims_reporting/medical_claim.rb:48]: I couldn't get this COALESCE to work so I just pulled it out

File: drivers/claims_reporting/app/models/claims_reporting/medical_claim.rb [hunk 2]
```diff
@@ -9,6 +9,9 @@ module ClaimsReporting
   class MedicalClaim < HealthBase
     extend Memoist
 
+    ANNUAL_WELLCARE_PROCEDURE_CODES = [(99381...99387).to_a, (99391...99397).to_a, 99461, 'G0438', 'G0439', 'T1015'].flatten.map(&:to_s).freeze
+    ANNUAL_WELLCARE_DIAGNOSIS_CODE_PREFIXES = ['Z00', 'Z02', 'Z76'].freeze
```
Comment (gigxz) [inline drivers/claims_reporting/app/models/claims_reporting/medical_claim.rb:13]: @awisema do you think it's safe to include everything with these prefixes, or should it be more granular for Z02 and Z76?

Comment (awisema) [reply drivers/claims_reporting/app/models/claims_reporting/medical_claim.rb:13]: I think its fine, but if you are worried, there is enough data at this point, that if you want, we can run the SQL and see if there looks to be anything they don't include...

File: app/models/health/agency_performance.rb [hunk 3]
```diff
@@ -324,6 +326,17 @@ def qa_signature_dates
         pluck(:patient_id).uniq
     end
 
+    private def with_annual_well_care_visit_in_12_months
+      # range is a 12 month period ending 3 months ago, because claims data is always 3 months behind
+      service_range = (Date.current - 15.months...Date.current - 3.months)
```
Comment (awisema) [inline app/models/health/agency_performance.rb]: I think we can simplify this to just being the last 12 months...

And then complicate the view, by determining the newest service date in the range, and reporting that as the end date for the calculation.

**Review by awisema — CHANGES_REQUESTED** [review]

**Review by awisema — APPROVED** [review]

**Pull request merged**